node/src/components/linear_chain.rs

```diff
@@ -148,43 +150,35 @@ where
     ) -> Effects<Self::Event> {
         match event {
             Event::Request(LinearChainRequest::BlockRequest(block_hash, sender)) => async move {
-                match effect_builder.get_block_from_storage(block_hash).await {
-                    None => debug!("failed to get {} for {}", block_hash, sender),
-                    Some(block) => match Message::new_get_response(&block) {
-                        Ok(message) => effect_builder.send_message(sender, message).await,
-                        Err(error) => error!("failed to create get-response {}", error),
-                    },
+                let fetched_or_not_found_block =
+                    match effect_builder.get_block_from_storage(block_hash).await {
+                        None => FetchedOrNotFound::NotFound(block_hash),
+                        Some(block) => FetchedOrNotFound::Fetched(block),
+                    };
```

Comment:
I think this pattern got repeated a few times. Would it be possible to extract something like `FetchedOrNotFound::from_option()`?